src/service.c

```diff
@@ -330,14 +330,21 @@ static int lredirect(svc_t *svc)
 
 				snprintf(rot, sizeof(rot), "%d:%d", logfile_size_max, logfile_count_max);
 				snprintf(pid, sizeof(pid), "%d", svc_pid);
-				execlp("logger", "logger", "-f", svc->log.file, "-b", "-t", tag, "-p", prio, debug ? "-s" : "", "-I", pid, "-r", rot, NULL);
+				if (debug)
+					execlp("logger", "logger", "-f", svc->log.file, "-b", "-t", tag, "-p", prio, "-s", "-I", pid, "-r", rot, NULL);
+				else
+					execlp("logger", "logger", "-f", svc->log.file, "-b", "-t", tag, "-p", prio, "-I", pid, "-r", rot, NULL);
```

Comment:
Great find!

However, isn't a better solution to just move the optional `-s` to the end?  E.g.,

```C
execlp("logger", "logger", "-f", svc->log.file, "-b", "-t", tag, "-p", prio, "-I", pid, "-r", rot, debug ? "-s" : NULL, NULL);
```

Please change to that for all these I'll be more than happy to merge the PR.

Reply:
Didn't even think of that, much simpler. Changed.